client/api/go-client/client.go

```diff
@@ -48,11 +63,49 @@ func NewClient(host, user, key string) *Client {
 	return c
 }
 
+func NewClientTLS(host, user, key string, tlsOpts *ClientTLSOptions) (*Client, error) {
+	c := NewClient(host, user, key)
+	if err := c.SetTLSOptions(tlsOpts); err != nil {
```

Comment:
do we need to check host schema (whether its HTTP or HTTPS before creating HTTPS client)

Reply:
I don't think so. Internally, the Go http client will use the protocol indicated by the url, but we don't need to key to conditionalize our logic. If HTTPS is configured but not used by the request these options will just be ignored by the http client code.